service/history/workflow/mutable_state_impl.go

```diff
@@ -8085,6 +8087,10 @@ func (ms *MutableStateImpl) StartDeploymentTransition(deployment *deploymentpb.D
 		ms.GetExecutionInfo().VersioningInfo = versioningInfo
 	}
 
+	if ms.GetEffectiveDeployment().Equal(deployment) {
+		return serviceerror.NewInternal("start transition should receive a version different from effective version")
```

Comment:
can you put the strings of each version in this error?

Reply:
(or do you expect it to be never hit, in which case maybe that's not worth the additional CI run)